cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/RepositoryManagerTest.java

```diff
@@ -157,4 +170,129 @@ public void testValidateSuccess() throws IOException {
     SourceControlConfig sourceControlConfig = new SourceControlConfig(new NamespaceId(NAMESPACE), config, cConf);
     RepositoryManager.validateConfig(secureStore, sourceControlConfig);
   }
+
+  @Test
+  public void testCommitAndPushSuccess() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig config = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+      .setLink(serverURL + "ignored")
+      .setDefaultBranch("develop")
+      .setAuthType(AuthType.PAT)
+      .setTokenName(GITHUB_TOKEN_NAME)
+      .build();
+    SourceControlConfig sourceControlConfig = new SourceControlConfig(new NamespaceId(NAMESPACE), config, cConf);
+
+    try (RepositoryManager manager = new RepositoryManager(secureStore, sourceControlConfig)) {
+      manager.cloneRemote();
+      CommitMeta commitMeta = new CommitMeta("author", "committer", 100, "message");
+      Path filePath = manager.getBasePath().resolve("file1");
```

Comment:
Based on https://github.com/cdapio/cdap/pull/14917#discussion_r1106272625, we should use relative paths. There will be a change in the push method also, since git expects the file pattern to be relative to the directory where the .git folder is there while `manager.getBasePath()` can have an optional path prefix configured through the repository config. See getFileHash method implementation.
We can add a TODO to use relative paths do that once the fileHash PR is merged.